File: src/main/java/io/github/chaosdave34/kitpvp/items/impl/engineer/TurretItem.java
package io.github.chaosdave34.kitpvp.items.impl.engineer;

import io.github.chaosdave34.kitpvp.abilities.Ability;
import io.github.chaosdave34.kitpvp.abilities.AbilityHandler;
import io.github.chaosdave34.kitpvp.items.CustomItem;
import net.kyori.adventure.text.Component;
import org.bukkit.Material;
import org.jetbrains.annotations.NotNull;

import java.util.List;

public class TurretItem extends CustomItem {

    public TurretItem() {
        super(Material.STRAY_SPAWN_EGG, "turret", false, true);
    }

    @Override
    public @NotNull Component getName() {
        return createSimpleItemName("Turret");
    }

    @Override
    public @NotNull List<Ability> getAbilities() {
        return List.of(AbilityHandler.TURRET);
    }
}

File: src/main/kotlin/io/github/chaosdave34/kitpvp/items/impl/engineer/TurretItem.kt
package io.github.chaosdave34.kitpvp.items.impl.engineer

import io.github.chaosdave34.kitpvp.abilities.Ability
import io.github.chaosdave34.kitpvp.abilities.AbilityHandler
import io.github.chaosdave34.kitpvp.items.CustomItem
import org.bukkit.Material

class TurretItem : CustomItem(Material.FIREWORK_ROCKET, "turret","Turret",  preventPlacingAndUsing = true) {
    override fun getAbilities(): List<Ability> = listOf(AbilityHandler.TURRET)
}